### DISCUSSION TITLE
How to access a plugin?

### DISCUSSION THREAD

**Discussion by github-reader** [hairyhenderson/gomplate#1371] [Q&A] [answered]
Hello

I am working with AWS and have defined a plugin in .gomplate.yaml.

```
plugins:
  awsAccount:
    cmd: curl 169.254.169.254/latest/meta-data/identity-credentials/ec2/info 2>/dev/null | jq '.AccountId'

```
How can I use it in a template? With a datasource I would use the function datasource. Passing the plugin via command line is not an option.

Normally I would use the function aws.Account. But this function fails with an STS timeout error in 1 region. I am using multiple regions. So I must be able to specify a timeout. Plugins allow this.
AWS error message.
` ... at <aws.Account>: error calling Account: RequestError: send request failed\ncaused by: Post \"https://sts.amazonaws.com/\": context deadline exceeded (Client.Timeout exceeded while awaiting headers)"`

**Comment by hairyhenderson** [comment, marked as answer]
If you're having timeouts, maybe have a look at adjusting the `AWS_TIMEOUT` environment variable - see [_Configuring AWS_](https://docs.gomplate.ca/functions/aws/#configuring-aws) for details.

As far as using plugins, [the `plugins` docs](https://docs.gomplate.ca/config/#plugins) should cover usage. In short, it makes a new function available with the same name that you use in your config. In your case that would be `awsAccount`.

As a general note, since you're piping output in your particular command you'll probably need to put that in a shell script, or at least interpret it with `sh` (like `cmd: sh -c "curl 169.254.169.254/latest/meta-data/identity-credentials/ec2/info 2>/dev/null | jq '.AccountId'"`), because pipes (`2>` and `|` in your command) are shell-specific constructs.

Hope that helps!

**Reply by github-reader** [reply]
Thank you very much for your reply and your work. Setting AWS_TIMEOUT=5000 solved my problem.
